File: 2021/j3.py
"""
Five digits

0, 1
+odd = left
+even, not 0 = right
+0 = same as previous

2, 3, 4
"""

instructions = []
last_direction = None

while True:
    directions = input()

    if directions == "99999":
        break

    add = int(directions[0]) + int(directions[1])

    if add == 0:
        direction = last_direction
    elif add % 2 == 0:
        direction = "right"
    else:
        direction = "left"

    last_direction = direction

    instructions.append(f"{direction} {directions[-3:]}")

for instruction in instructions:
    print(instruction)

File: 2021/j3.cpp
#include <vector>
#include <iostream>

using namespace std;


struct Instruction {
  bool direction;
  int steps;
};

int main() {

  // false = left
  // true = right
  bool lastInstruction = false;
  vector<Instruction> instructions;

  while (true) {

    string instruction;
    cin >> instruction;

    if (instruction == "99999") {
      break;
    }

    int sum = (instruction[0] - '0') + (instruction[1] - '0');
    int steps = stoi(instruction.substr(2,3));

    if (sum % 2 == 0 && sum != 0) {
      lastInstruction = true;
    } else if (sum % 2 != 0) {
      lastInstruction = false;
    }

    instructions.push_back({ lastInstruction, steps });

  }

  for (int i = 0; i < instructions.size(); i++) {
    if (instructions[i].direction) {
      cout << "right " << instructions[i].steps << endl;
    } else {
      cout << "left " << instructions[i].steps << endl;
    }
  }

}

